src/execution_plan/ops/op_sort.c

```diff
@@ -0,0 +1,215 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#include "op_sort.h"
+
+#include "../../util/arr.h"
+#include "../../util/qsort.h"
+#include "../../util/rmalloc.h"
+
+static bool _record_islt(Record a, Record b, const Sort *op) {    
+    // First N values in record correspond to RETURN expressions
+    // N .. M values correspond to ORDER-BY expressions
+    // Query: RETURN A.V, B.V, C.V ORDER BY B.W, C.V*A.V
+    // Record
+    // ------------------------------------
+    // |  V0  |  V1  |  V2  |  V3  |  V4  |
+    // ------------------------------------
+    // V0 - A.V
+    // V1 - B.V
+    // V2 - C.V
+    // V3 - B.W
+    // V4 - C.V*A.
+    uint offset = array_len(op->ast->returnNode->returnElements);
+    uint comparables = array_len(op->ast->orderNode->expressions);
+
+    for(uint i = 0; i < comparables; i++) {
+        SIValue aVal = Record_GetScalar(a, offset+i);
+        SIValue bVal = Record_GetScalar(b, offset+i);
+        int rel = SIValue_Order(aVal, bVal);
```

Comment:
There's an interesting case when the ORDER entity resolves to a pointer, such as:
`MATCH (a:person) RETURN a ORDER BY a`
Neo will order by ID in this case. For us, `rel` will always return 0, so the query doesn't fail, but direction is ignored and no order is imposed.
This isn't a particularly high priority, but with SKIP and LIMIT this would be a reasonable way to paginate results. Maybe upstream we can replace the `ORDER BY (a)` with an `ORDER BY ID(a)` or something like that.

Reply:
I think we should take into account the value type when comparing, 
1. we should introduce two new types to SIValue, Node and edge.
2. when comparing two graph entities, compare using their ID.